### ISSUE TITLE
Noise cancellation requires allowing `blob:` in CSP

### ISSUE THREAD

**Issue by Dosant** [twilio/twilio-video.js#1880]
After upgrading to 2.24.1, noise cancellation stopped working for us. 

When trying to load the noise cancellation code, there is a CSP-related error and the noise cancellation code fails to load: 

```
dynamicimport.js:40 Refused to load the script 'blob:http://localhost:4200/f39c03f4-9122-4e7f-b7ed-fb08f96dc102' because it violates the following Content Security Policy directive: "script-src 'self'". Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.
```

Looks like this is caused by the workaround introduced in https://github.com/twilio/twilio-video.js/pull/1872, and to make the feature work with this workaround, we'll have to loosen up the CSP to allow `blob:` which could potentially increase the attack surface. 

Note that 2.24.0 worked fine in our environment without [the import workaround](https://github.com/twilio/twilio-video.js/pull/1872).

I suggest the team to consider if it is worth keeping the workaround considering this CSP downside.

**Comment by PikaJoyce** [comment]
Hi @Dosant,

Thank you for opening this issue with us. I'll bring this up to the team and will respond on this thread as soon as we have any updates! 

Best,
Joyce

**Comment by Dosant** [comment]
> Thank you for opening this issue with us. I'll bring this up to the team and will respond on this thread as soon as we have any updates!

Thank you!

> However, I do have a question. Are you using the relative path for the sdk assets? Or are you trying to load the krisp SDK from another domain?

relative path. the CSP error happens it even tries to download krisp. It happens when it tries to execute script from blob

**Comment by PikaJoyce** [comment]
Hey @Dosant,

Gotcha, thanks for your response. I've created a an internal ticket (VIDEO-11537) to track this issue! I'll continue to post updates to this thread as we progress through our investigations. 

Thank you for your patience!

Best,
Joyce

**Comment by PikaJoyce** [comment]
Hi @Dosant, 

The team has begun investigations for this. I was wondering if it was possible for you to provide an example of this happening as I haven't been able to reproduce this issue across several applications. Thanks!

Best,
Joyce

**Comment by Dosant** [comment]
@PikaJoyce, to reproduce this, you'll need an app that has a [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) configured. For example, the one where all content can come only from the site's own origin: 

```
Content-Security-Policy: default-src 'self'
```

The problem with the current noise cancelation loading method is that it requires "weakening" the content security policy, which possibly could open up an attack vector: 

```
Content-Security-Policy: default-src 'self'; script-src blob:
```

**Comment by manjeshbhargav** [comment]
@Dosant ,

I investigated a bit more, and have come up with a potential workaround. The only caveat is that our noise suppression vendor library requires the `unsafe-eval` CSP directive to be enabled. So before I go ahead with the change, I wanted to ask you if adding `Content-Security-Policy: default-src 'self' 'unsafe-eval'` is acceptable to you.

**Comment by manjeshbhargav** [comment]
Hi @Dosant ,

I have released 2.24.2, which removes the blob requirement. Thanks for bringing this to my notice.
